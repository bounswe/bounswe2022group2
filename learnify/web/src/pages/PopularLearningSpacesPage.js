import React, { useState, useEffect } from 'react';
import axios from 'axios';
import './style.css';
import LearningSpaceDetailBox from '../components/LearningSpaceDetailBox';
import Footer from '../components/Footer';
import NavBar from '../components/NavBar';

function PopularLearningSpacesPage() {

    const [learningspaces, setLearningspaces] = useState([])

    const token = localStorage.getItem('token');

    useEffect(() => {
        const getPopularLearningSpaces = async () => {
            const res = await axios.get(`${process.env.REACT_APP_BACKEND_BASE_URL}learningspace`, {
                headers: {
                    'Content-type': 'application/json; charset=UTF-8',
                    'Authorization': `${token}`,
                }
            });
            setLearningspaces(res.data.learning_spaces);
        }
        getPopularLearningSpaces();
    }, []);

    const lsBoxes = learningspaces.slice(0, 10).map(ls => (
        <LearningSpaceDetailBox title={ls.title} description={ls.description} icon_id={ls.icon_id} num_participants={ls.num_participants} url={ls._id}/>
    ));

    return (
        <div>
            <NavBar/>
            <div className="popularlspage-container">
                <div className="popularlspage-wrapper">
                    <div className="popularlspage-title">
                        <h1>Browsing the current most popular learning spaces:</h1>
                    </div>
                </div>
                <div className="popularlspage-body" data-testid="popularls-body">
                    {lsBoxes}
                </div>
            </div>
            <Footer/>
        </div>
    )
}

export default PopularLearningSpacesPage;