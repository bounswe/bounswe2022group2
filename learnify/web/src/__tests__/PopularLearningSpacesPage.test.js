import React from "react";
import PopularLearningSpacesPage from "../pages/PopularLearningSpacesPage";
import { render, screen } from '@testing-library/react';
import renderer from 'react-test-renderer';
import { BrowserRouter } from "react-router-dom";

describe("PopularLearningSpacesPage", () => {
    test('should render page layout correctly', async () => {
        const component = renderer.create(<BrowserRouter><PopularLearningSpacesPage /></BrowserRouter>).toJSON();
        expect(component).toMatchSnapshot();
    });
}   );

test('popular learning spaces page footer renders correctly', async () => {
    render(<BrowserRouter><PopularLearningSpacesPage /></BrowserRouter>);
    const footer = screen.getByTestId(/footer/i);
    expect(footer).toBeInTheDocument();
}   );

test('popular learning spaces page navbar renders correctly', async () => {
    render(<BrowserRouter><PopularLearningSpacesPage /></BrowserRouter>);
    const navbar = screen.getByTestId(/navbar/i);
    expect(navbar).toBeInTheDocument();
}   );

test('popular learning spaces page header renders correctly', async () => {
    render(<BrowserRouter><PopularLearningSpacesPage /></BrowserRouter>);
    const header = screen.getByText(/Browsing the current most popular learning spaces:/i);
    expect(header).toBeInTheDocument();
}   );

test('popular learning spaces page body renders correctly', async () => {
    render(<BrowserRouter><PopularLearningSpacesPage /></BrowserRouter>);
    const body = screen.getByTestId(/popularls-body/i);
    expect(body).toBeInTheDocument();
}   );

