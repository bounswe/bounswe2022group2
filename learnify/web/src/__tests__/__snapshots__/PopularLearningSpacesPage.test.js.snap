// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PopularLearningSpacesPage should render page layout correctly 1`] = `
<div>
  <nav
    className="nav"
    data-testid="navbar"
  >
    <a
      href="/home"
    >
      <img
        alt="Learnify Logo Navbar"
        height={60}
        src="logo-dblue.png"
      />
    </a>
    <ul>
      <div
        className="relative"
      >
        <input
          className="search-input-field"
          onChange={[Function]}
          placeholder="Search learning spaces"
          size={30}
          type="text"
          value=""
        />
        <div
          className="search-results-container"
        />
      </div>
      <li>
        <a
          className="navBarText"
          href="/notifications"
        >
          <img
            alt="Notifications Icon"
            height={20}
            src="notification-icon.svg"
          />
        </a>
      </li>
      <li>
        <a
          className="navBarText"
          href="/home"
        >
          Home
        </a>
      </li>
      <li>
        <a
          className="navBarText"
          href="/categories"
        >
          Categories
        </a>
      </li>
      <li>
        <a
          className="navBarText"
          href="/about"
        >
          About
        </a>
      </li>
      <li>
        <a
          className="navBarText"
          href="/contact"
        >
          Contact
        </a>
      </li>
      <li>
        <a
          className="navBarText"
          href="/profile/null"
        >
          Profile
        </a>
      </li>
    </ul>
  </nav>
  <div
    className="popularlspage-container"
  >
    <div
      className="popularlspage-wrapper"
    >
      <div
        className="popularlspage-title"
      >
        <h1>
          Browsing the current most popular learning spaces:
        </h1>
      </div>
    </div>
    <div
      className="popularlspage-body"
      data-testid="popularls-body"
    />
  </div>
  <div
    className="footer-parent-layout"
    data-testid="footer"
  >
    <div
      className="footer-container"
    >
      <div
        className="footer-row"
      >
        <div
          className="footer-column-l"
        >
          <div
            className="footer-item-bold"
          >
            Learnify
          </div>
          <div
            className="footer-item-reg"
          >
            is a learning space for everyone, where learning is 
            <br />
             achieved through collaboration.
          </div>
        </div>
        <div
          className="footer-column-r"
        >
          <a
            className="footer-item-bold"
            href="/home"
          >
            Home
          </a>
          <a
            className="footer-item-bold"
            href="/categories"
          >
            Categories
          </a>
          <a
            className="footer-item-bold"
            href="/notifications"
          >
            Notifications
          </a>
        </div>
        <div
          className="footer-column-r"
        >
          <a
            className="footer-item-bold"
            href="/profile"
          >
            Profile
          </a>
          <a
            className="footer-item-bold"
            href="/contact"
          >
            Contact
          </a>
          <a
            className="footer-item-bold"
            href="/about"
          >
            About
          </a>
        </div>
        <div
          className="footer-column-r"
        >
          <a
            className="footer-item-bold"
            href="/user/learning-spaces"
          >
            My Learning Spaces
          </a>
          <a
            className="footer-item-bold"
            href="/popular-learning-spaces"
          >
            Popular Learning Spaces
          </a>
          <a
            className="footer-item-bold"
            href="/recommended-learning-spaces"
          >
            Recommended Learning Spaces
          </a>
        </div>
        <div
          className="footer-column-r"
        >
          2022
           Â© Learnify Team. All rights reserved.
        </div>
      </div>
    </div>
  </div>
</div>
`;
